walletunlocker/service_test.go

```diff
@@ -434,7 +433,7 @@ func TestChangeWalletPasswordNewRootkey(t *testing.T) {
 	// requested.
 	var tempFiles []string
 	for i := 0; i < 3; i++ {
-		file, err := ioutil.TempFile(testDir, "")
+		file, err := os.CreateTemp(testDir, "")
```

Comment:
Hmm, this doesn't seem to fix the flaky failure I mentioned before. It looks like you uncovered a problem with the test itself.
We need to add `require.NoError(t, unlockMsg.UnloadWallet())` on Line 608 (directly after `service.MacResponseChan <- testMac`) to fix the flake.